pkg/image/nbdkit.go

```diff
@@ -240,10 +240,18 @@ func (n *Nbdkit) StartNbdkit(source string) error {
 
 // Default nbdkit log watcher, just logs lines as nbdkit prints them.
 func watchNbdLog(output *bufio.Reader) {
+	f, err := os.Create("/tmp/nbdkit.log")
```

Comment:
Ddescription states "Default nbdkit log watcher, just logs lines as nbdkit prints them."  but now this also creates its own f log file.

Can you tell me why this log is needed? It is stored on the pod where nbd is run, am I right? And the pod is only running for the time it takes operation to complete so there is no problem with managing the log size.

Reply:
Yes, there is no problem with managing the log size. It's a few lines at the importer pod, which is cleaned up automatically when the importer completes.
I wanted the messages to be part of the termination message in a failure, but didn't want to stop printing them as they happen, in case some useful message appears at success.
There didn't seem to be a great way to run this process, monitor its output, and receive the updated output when something else terminates unsuccessfully. I also didn't see a way to make nbdkit log to two places.

Reply:
So looking at the code, we are writing to /tmp/nbdkit.log but then never use it for anything as far as I can tell. Am I missing something?

Reply:
[Used here](https://github.com/kubevirt/containerized-data-importer/pull/1818/files#diff-911dad1f9fe8bfd9b383cded047f7472b0b56ad8b201ac8a3f6fad28f00605ab)